### PR TITLE
Change Theme to a DropDown with flyout 

### PR DESCRIPTION
**Pull request by Natestah**
- Little bit of MVVP in here to enable bindings in the AXAML
- Drop down, allows me to look at specific themes, As I intend to leverage a little more of the info from the TextMateSharp JSON's

HELLO, I'm the author of BlitzSearch ( https://natestah.com ).  I'm making heavy use of AvaloniaEdit and there are things that I want to do with it.  This would be a small Ice Breaker change, would be awesome if the efforts could align.

### CODE REVIEW

File: src/AvaloniaEdit.Demo/MainWindow.xaml [hunk 1]
```diff
@@ -23,7 +25,21 @@
             <Button Name="addControlBtn" Content="Add Button" VerticalAlignment="Stretch" VerticalContentAlignment="Center"/>
             <Button Name="clearControlBtn" Content="Clear Buttons" VerticalAlignment="Stretch" VerticalContentAlignment="Center"/>
             <ComboBox Name="syntaxModeCombo" VerticalAlignment="Stretch" VerticalContentAlignment="Center"/>
-            <Button Name="changeThemeBtn" Content="Change theme" VerticalAlignment="Stretch" VerticalContentAlignment="Center"/>
+            
+            <DropDownButton Content="Change theme" VerticalAlignment="Stretch" VerticalContentAlignment="Center">
+                <Button.Flyout>
+                    <Flyout>
+                        <ListBox ItemsSource="{Binding AllThemes }" SelectedItem="{Binding SelectedTheme}">
```
Comment (maxkatz6) [inline src/AvaloniaEdit.Demo/MainWindow.xaml:32]: It can be MenuFlyout with ToggleType=Radio, in 11.1 Avalonia.

**Review by maxkatz6 — APPROVED** [review]

File: src/AvaloniaEdit.Demo/MainWindow.xaml (hunk 1 above)
Comment (Natestah) [reply src/AvaloniaEdit.Demo/MainWindow.xaml:32]: The toggle type radio is pretty awesome, I'm using it in Blitz Search but sure if that's something we want with this library? since 11.1 is all marked rc1,beta, etc.

File: src/AvaloniaEdit.TextMate/AvaloniaEdit.TextMate.csproj [hunk 2]
```diff
@@ -12,8 +12,9 @@
 
   <ItemGroup>
     <PackageReference Include="Avalonia" Version="$(AvaloniaVersion)" />
-    <PackageReference Include="TextMateSharp" Version="$(TextMateSharpVersion)" />
-    <PackageReference Include="TextMateSharp.Grammars" Version="$(TextMateSharpVersion)" />
+    <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.0" />
```
Comment (maxkatz6) [inline src/AvaloniaEdit.TextMate/AvaloniaEdit.TextMate.csproj:15]: Can we revert these changes from non-demo project?
Controls shouldn't depend on a single specific theme.

File: src/AvaloniaEdit.TextMate/TextMate.cs [hunk 3]
```diff
@@ -61,17 +70,93 @@ public void SetGrammar(string scopeName)
...
 
-                GetOrCreateTransformer().SetTheme(_textMateRegistry.GetTheme());
+                var registryTheme = _textMateRegistry.GetTheme();
+                GetOrCreateTransformer().SetTheme(registryTheme);
 
                 _tmModel?.InvalidateLine(0);
 
                 _editorModel?.InvalidateViewPortLines();
+
+                if (Application.Current?.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime
```
Comment (maxkatz6) [inline src/AvaloniaEdit.TextMate/TextMate.cs:96]: It's a bad practice to depend on IClassicDesktopStyleApplicationLifetime.

File: src/AvaloniaEdit.TextMate/TextMate.cs [hunk 4]
```diff
@@ -61,17 +70,93 @@ public void SetGrammar(string scopeName)
...
+                    {
+                        MainWindow: Window window
+                    })
+                {
+                    _guiColorDictionary = registryTheme.GetGuiColorDictionary();
+                    var themeName = theme.GetName();
+
+                    // Applies to the application on a whole.. Some might want to opt out of this for their own setups.
+                    if (applyWindowProperties)
```
Comment (maxkatz6) [inline src/AvaloniaEdit.TextMate/TextMate.cs:105]: I am not sure if I understand these changes, but why would control change global theme?
If text editor has custom theming solution, it should be contained by this control.

File: src/AvaloniaEdit.TextMate/TextMate.cs [hunk 5]
```diff
@@ -61,17 +70,93 @@ public void SetGrammar(string scopeName)
...
+                        _editor.LineNumbersForeground = _editor.Foreground;
+                    }
+                    
+                    AppliedTheme?.Invoke(this,this);
+                }
             }
+            
+            //These come out of CurrentHighlightRenderer sealed class. 
+            public static readonly Color CurrentHighlightRendererDefaultBackground = Color.FromArgb(22, 20, 220, 224);
+            public static readonly Color CurrentHighlightRendererDefaultBorder = Color.FromArgb(52, 0, 255, 110);
```
Comment (maxkatz6) [inline src/AvaloniaEdit.TextMate/TextMate.cs:159]: These hardcoded colors should be avoided and removed in CurrentLineHighlightRenderer class as well. Colors should be defined by XAML themes. But that's for another day, I guess.

**Review by maxkatz6 — CHANGES_REQUESTED** [review]

**Comment by danipen** [comment]
@Natestah I tried the changes and found some issues:
* Once you open the demo window for the first time, the selection is not visible or it doesn't work.
* After changing the theme, the selection becomes visible.
* After changing the theme again, the selection color doesn't properly changes.

See the attached video for clarification:

https://github.com/AvaloniaUI/AvaloniaEdit/assets/501613/27b92e3c-6d86-491e-9318-fb2675dbeb3b

File: src/AvaloniaEdit.TextMate/TextMate.cs (hunk 5 above)
Comment (Natestah) [reply src/AvaloniaEdit.TextMate/TextMate.cs]: I at least got rid of the Duplicate definition,  but I agree, this should be some property up higher on the textview, not in the background highlight renderer. Just not today

File: src/AvaloniaEdit.TextMate/TextMate.cs (hunk 3 above)
Comment (Natestah) [reply src/AvaloniaEdit.TextMate/TextMate.cs]: Still getting my head around some of the Avalonia stuff, but this wasn't the place for this. it's now living in the Demo apps app.axaml.cs

File: src/AvaloniaEdit.TextMate/TextMate.cs (hunk 4 above)
Comment (Natestah) [reply src/AvaloniaEdit.TextMate/TextMate.cs]: removed..

**Comment by Natestah** [comment]
I think Everything is up-to-date, thank you for time and patience. I'm still learning how to use GitHub and Avalonia

**Review by maxkatz6 — APPROVED** [review]
LGTM, thank you.

**Comment by maxkatz6** [comment]
cc @danipen if you want to have another look

**Comment by danipen** [comment]
I'm rethinking this a bit. 

We should let users use this functionality, but it shouldn't be mandatory. Some customers likely have custom background or selection colors, so we'd be breaking their styles. Setting styles for the text editor should be the user's responsibility. 

The TextMate API should be responsible of notifying about theme changes and providing the theme colors. We might also include constants to help the users to easily get the colors. Let me rethink this a bit and I'll do a small refactor.

File: src/AvaloniaEdit.TextMate/TextTransformation.cs [hunk 6]
```diff
@@ -18,7 +18,7 @@ public class ForegroundTextTransformation : TextTransformation
         public Action<Exception> ExceptionHandler { get; set; }
         public int ForegroundColor { get; set; }
         public int BackgroundColor { get; set; }
-        public int FontStyle { get; set; }
+        public FontStyle FontStyle { get; set; }
```
Comment (maxkatz6) [inline src/AvaloniaEdit.TextMate/TextTransformation.cs:21]: I haven't noticed before, but changing public property type is an ABI breaking change. Might not be critical, but needs to be carefully considered.

Comment (danipen) [reply src/AvaloniaEdit.TextMate/TextTransformation.cs:21]: Yes, this is an API breaking from TextMateSharp ...
